### ISSUE TITLE
ZipException: Expected central directory entry not found (#1)

### ISSUE THREAD

**Issue by dknchris** [srikanth-lingala/zip4j#290]
One of my users is getting this error when the app is calling ZipFile.getFileHeaders(). I've attached the zip file for your reference. The zip was created using zip4j v2.7.0 on the user's Android device.
[error.zip](https://github.com/srikanth-lingala/zip4j/files/6054468/error.zip)

I've also tested extracting it with 7z and it works fine. Also tried the 'Test archive' option in 7z and it reports that there are no errors in the archive.

Can you try it out at your end? Please let me know if you need more information.

```
net.lingala.zip4j.exception.ZipException: Expected central directory entry not found (#1)
        at net.lingala.zip4j.headers.HeaderReader.readCentralDirectory(HeaderReader.java:153)
        at net.lingala.zip4j.headers.HeaderReader.readAllHeaders(HeaderReader.java:101)
        at net.lingala.zip4j.ZipFile.readZipInfo(ZipFile.java:1136)
        at net.lingala.zip4j.ZipFile.getFileHeaders(ZipFile.java:700)
```

**Comment by srikanth-lingala** [comment]
Issue identified. Fix in progress.

**Comment by ucchino** [comment]
Thank you for the work you are doing. I wanted to ask you for an info on this issue. I see you're working on it. Strangely i have this error on mac but not on windows.

When do you think it will be ready? even if release snapshot?

Best regards

Paolo

**Comment by bobvandenberge** [comment]
@srikanth-lingala we are also running into this issue on production. Do you perhaps have an indication when there will be a release with this fix included?

**Comment by srikanth-lingala** [comment]
@ucchino @bobvandenberge I will try to get a fix out by this weekend. Free time is hard to find nowadays with lockdowns and all (work, kids at home, etc). But I understand your position, and will work on this fix in the next days.

**Comment by srikanth-lingala** [comment]
Issue fixed. I will include the fix in the next release which should be in a day or two. I will update this thread again once the new version is available

**Comment by srikanth-lingala** [comment]
Fixed in v2.8.0 released today.
